x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -80,6 +80,29 @@ static DeprecationIssue httpPipeliningCheck(List<NodeInfo> nodeInfos, List<NodeS
         return null;
     }
 
+    static DeprecationIssue discoveryConfigurationCheck(List<NodeInfo> nodeInfos, List<NodeStats> nodeStats) {
+        if (nodeInfos.size() == 1
+            && nodeInfos.stream().anyMatch(nodeInfo -> "single-node".equals(nodeInfo.getSettings().get("discovery.type")))) {
```

Comment:
Technically this check only applies if `discovery.type` is `zen2`:

https://github.com/elastic/elasticsearch/blob/cd1bec3a06df97a5ec7c135f475d0059dc1bee64/server/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java#L728-L730

Of course `zen2` isn't a thing in `6.x` so we can't check for that; I think we should check for usage of the default discovery type. We have already changed the default to `zen2` in `master` and we hope to deprecate `zen` as a valid discovery type, which will I guess need another deprecation check.

FYI the comparison with `single-node` discovery applies to all bootstrap checks:

https://github.com/elastic/elasticsearch/blob/cd1bec3a06df97a5ec7c135f475d0059dc1bee64/server/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java#L187

In practice I think we expect `single-node` and `zen2` to be the only options here, rendering these two statements practically equivalent, but that's still to be confirmed and may change in future.

Also please could you use `DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey()` instead of the literal setting key. It makes it that bit easier to find usages of a setting in the IDE, and that bit less likely that we miss this usage in some future work.